app/models/course.rb

```diff
@@ -157,9 +157,10 @@ def new_draft_attributes
                                                  .except(:json_data)
 
         latest_published_enrichment_attributes[:status] = :draft
+        latest_published_enrichment_attributes[:last_published_timestamp_utc] = nil
```

Comment:
Maybe to avoid bugs return the core attributes from the `if` and then merge in the overridden params once, eg

```ruby
new_enrichment_attributes = if latest_published_enrichment.present?
        latest_published_enrichment_attributes = latest_published_enrichment
                                                 .dup
                                                 .attributes
                                                 .with_indifferent_access
                                                 .except(:json_data)
  else
    {}
  end

new_enrichment_attributes.merge(
  { status: :draft, last_published_timestamp_utc: nil }
)
```

Also — could we use `current_published_enrichment` instead of `most_recent.published.first`?

Reply:
I've refactored it to use the above but `current_published_enrichment` breaks a handful of specs, so I'd like to spend a bit more time understanding the code and making sure logic and behaviour is being sufficiently tested before making further changes